### DISCUSSION TITLE
rAudio 20221117: Album data not available

### DISCUSSION THREAD

**Discussion by klassikmann** [rern/rAudio#1107] [Fixed Issues]
The first time after the upgrade to 20221117 I wanted to use the album view. But I got
<img src="https://user-images.githubusercontent.com/25383513/203113602-e7ff5c6b-a3dc-4bc1-8f78-c5c84e459e45.jpg" width=300>
Not very optimistic I tried the suggestion:
<img src="https://user-images.githubusercontent.com/25383513/203113937-1fe268f1-2bf0-4587-9e75-12e2ec41f3c8.jpg" width=300>
And as expected the problem persisted, no album view possible.

**Comment by cmh714** [comment]
I get this running the Thumbnail update, if it helps:

Cover Art Thumbnails
update done.
OK

Cover Art Thumbnails - update ...
/srv/http/bash/albumthumbnail.sh ""

.  Update Album Thumbnails ...

. No albums found in database.

**Reply by ceejayemm** [reply]
This might be a case of 'teaching your grandmother to suck eggs' but...

I have been having a similar issue after the Music Library share (in my case an NFS share from OMV 6.x) went 'missing' after a new SD card was flashed with i20221117.  It turned out that this version wasn't picking up the backup.gz file I had placed on the new card.  Using Restore on a copy of the same backup file from the (partially configured) System page however did do the restore and I got my Library back again. So check if your Music Library share is actually mounted properly and available to rAudio-1.

Chris

**Comment by klassikmann** [comment]
@cmh714  @ceejayemm 
I used a different approach to nail the problem down. I emptied the data/mpd directory followed by a reboot. After that rAudio is rescanning the file store to fill data/mpd. At the end all files are present again, but "album" is empty reflecting the message: "**Album** data not available".

@rern 
I see two alternative solutions to the problem:

- Case 1: If "album" is necessary => the rescan procedure for data/mpd is insufficient.

or
- Case 2: If "album" is no longer necessary => the trigger event to issue the error message is wrong.

**Reply by cmh714** [reply]
I actually dont care :) Since most of my 6TB of flac files is live music, and the majority doesnt come with graphics, its meaningless feature for me personally. I just try and report issues as they arise for others to be aware and for rern to eventually fix or ignore.

**Comment by klassikmann** [comment]
@rern 
fyi update 20221122 does not solve the problem.

**Comment by R101G** [comment]
Similar (?) problem with 20221122 here. The NTFS USB stick is not showing on the System page. I have reverted to 20221117, where it is mounted and working, but shows a red dot on the System page.

**Comment by klassikmann** [comment]
@R101G 
Look into the directory "/srv/http/data/mpd". If the file "album" has zero size then you are affected by the problem I described.

**Reply by R101G** [reply]
@klassikmann 
It only contains NAS files. If I did not have a NAS it would be empty.

**Comment by klassikmann** [comment]
@R101G 
```
rAudio:~  #  cd /srv/http/data/mpd
rAudio:/srv/http/data/mpd  #  ls -l
total 972
-rw-r--r-- 1 mpd  audio      0 Nov 22 08:54 album
-rw-r--r-- 1 mpd  audio  43438 Nov 22 08:54 albumartist
-rw-r--r-- 1 mpd  audio  47348 Nov 22 08:54 artist
-rw-r--r-- 1 mpd  audio   9661 Nov 22 08:54 composer
-rw-r--r-- 1 mpd  audio      0 Nov 22 08:54 conductor
-rw-r--r-- 1 mpd  audio    243 Nov 22 08:54 counts
-rw-r--r-- 1 mpd  audio    726 Nov 22 08:54 date
-rw-r--r-- 1 mpd  audio    237 Nov 22 08:54 genre
-rw-r--r-- 1 mpd  audio      0 Nov 22 08:54 latest
-rw-r--r-- 1 mpd  audio 870285 Nov 22 08:54 mpd.db
-rw-r--r-- 1 root root      30 Nov 22 08:55 mpdignorelist
```
As you see "album" is empty.

**Comment by rern** [comment]
@klassikmann,
@R101G,

[edited] Please try today's update.

**Comment by R101G** [comment]
@rern 
This hasn't fixed my problem. The USB stick is not mounted, or listed in storage list on System page. I will create a new bug report.

**Comment by klassikmann** [comment]
Neither the UPDATE nor the additional regular update solved the problem. /srv/http/data/mpd looks as before.
EDIT:
Except for "album" all other files are are refreshed.

**Comment by ceejayemm** [comment]
Another thought, for what its worth :-)

Could this be a permissions issue as my '/srv/http/data/mpd' folder is owned by http:http as are it contents and all items are at least read/executable by everybody. It is however a SharedData installation and my Album data file is populated.

RAConservatory:/srv/http/data/mpd  #  ls -lrt
total 328
-rwxr-xr-x 1 mpd  audio 185563 Oct 10 10:10 mpd.db
-rwxr-xr-x 1 http http       0 Oct 10 10:10 latest
-rwxr-xr-x 1 http http   29149 Oct 10 19:39 albumbyartist
-rwxr-xr-x 1 http http   29149 Oct 10 19:39 album
-rwxr-xr-x 1 http http    2561 Oct 10 19:39 albumartist
-rwxr-xr-x 1 http http   16863 Oct 10 19:40 artist
-rwxr-xr-x 1 http http   33605 Oct 10 19:40 composer
-rwxr-xr-x 1 http http      36 Oct 10 19:40 conductor
-rwxr-xr-x 1 http http     534 Oct 10 19:40 genre
-rwxr-xr-x 1 http http    2815 Oct 10 19:40 date
-rwxr-xr-x 1 http http     819 Oct 10 19:40 mpdignorelist
-rwxr-xr-x 1 http http     245 Oct 25 10:21 counts

Chris

**Comment by klassikmann** [comment]
@ceejayemm 
I think ceejayemm  has a different problem than me. My "album" file is empty.

**Comment by ceejayemm** [comment]
I think you misunderstand. I DON'T have any issue with my Library, it works exactly as I would expect. The information I posted was to give you some background from what appears to be a working installation and might help you get to the bottom of your issues.

Regards.

Chris

**Reply by klassikmann** [reply]
@ceejayemm

o.k. Now I understand.

**Comment by klassikmann** [comment]
The problem is solved.
It looks as if any of the last updates reset the value of "Buffer - Output" in the player menu from "16384" back to its default of "8192"*.  After returning to my custom size a full database update fills the file "album" again.

*To be cautious I cannot fully exclude that I changed by myself "Buffer - Output" accidentially.

**Reply by rern** [reply]
@klassikmann,

The buffer should have been incrementally set to handle the large Library.
Please try:
- Reset the buffer back to 8192
- Disable the buffer
- Empty `/srv/http/data/mpd/album`
- Get the UPDATE branch
- Update Library

The Album should be populated and the buffer should be set and enabled.

**Reply by klassikmann** [reply]
@rern 
I performed the first four steps:

- Reset the buffer back to 8192
- Disable the buffer
- Empty `/srv/http/data/mpd/album`
- Get the UPDATE branch

To my surprise, the album view was still present (why?). I made a reboot. The  album view was still present. 

Last step:

- Update Library

Finished after 110 minutes (quite normal).
Observation: **All** files in `/srv/http/data/mpd` belong to mpd:audio now (that's new).

**Reply by klassikmann** [reply]
Next observation:
A local database update (new album) lasts "indefinitely" (perhaps the time of a full database update).
EDIT: Definitive "indefinitely". Undone "UPDATE"
